connector/src/main/java/org/geysermc/connector/network/translators/java/JavaPluginMessageTranslator.java

```diff
@@ -28,53 +28,80 @@
 import com.github.steveice10.mc.protocol.packet.ingame.client.ClientPluginMessagePacket;
 import com.github.steveice10.mc.protocol.packet.ingame.server.ServerPluginMessagePacket;
 import com.google.common.base.Charsets;
+import com.nukkitx.protocol.bedrock.packet.EmotePacket;
 import org.geysermc.connector.common.AuthType;
 import org.geysermc.connector.network.session.GeyserSession;
 import org.geysermc.connector.network.translators.PacketTranslator;
 import org.geysermc.connector.network.translators.Translator;
+import org.geysermc.connector.utils.StringByteUtil;
 import org.geysermc.cumulus.Form;
 import org.geysermc.cumulus.Forms;
 import org.geysermc.cumulus.util.FormType;
 
+import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
 
 @Translator(packet = ServerPluginMessagePacket.class)
 public class JavaPluginMessageTranslator extends PacketTranslator<ServerPluginMessagePacket> {
     @Override
     public void translate(GeyserSession session, ServerPluginMessagePacket packet) {
         // The only plugin messages it has to listen for are Floodgate plugin messages
-        if (session.getRemoteAuthType() != AuthType.FLOODGATE) {
-            return;
+
+        switch (packet.getChannel()) {
+            case "minecraft:register":
+                session.registerDownstreamPluginChannels(StringByteUtil.bytes2strings(packet.getData()));
+                break;
+            case "minecraft:unregister":
+                session.unregisterDownstreamPluginChannels(StringByteUtil.bytes2strings(packet.getData()));
+                break;
+            case "floodgate:form":
+                if (session.getRemoteAuthType() == AuthType.FLOODGATE) {
+                    handleFloodgateMessage(session, packet);
+                }
+                break;
+            case "geyser:emote":
+                handleEmote(session, packet);
+                break;
+            default:
+                //TODO feature: Here we should have a callback for extensions to handle their own plugin messages
         }
+    }
 
-        String channel = packet.getChannel();
+    private void handleEmote(GeyserSession session, ServerPluginMessagePacket packet) {
+        EmotePacket emotePacket = new EmotePacket();
+        ByteBuffer byteBuffer = ByteBuffer.wrap(packet.getData());
+        byte[] idBytes = new byte[byteBuffer.get()];
+        byteBuffer.get(idBytes);
+        emotePacket.setEmoteId(new String(idBytes, StandardCharsets.UTF_8));
+        emotePacket.setRuntimeEntityId(session.getEntityCache().getEntityByJavaId(byteBuffer.getLong()).getGeyserId());
+        session.sendUpstreamPacket(emotePacket);
```

Comment:
Think this looks better broken up, maybe don't need the middle new-line.
```suggestion
        EmotePacket emotePacket = new EmotePacket();
        
        ByteBuffer byteBuffer = ByteBuffer.wrap(packet.getData());
        byte[] idBytes = new byte[byteBuffer.get()];
        byteBuffer.get(idBytes);
        
        
        emotePacket.setEmoteId(new String(idBytes, StandardCharsets.UTF_8));
        emotePacket.setRuntimeEntityId(session.getEntityCache().getEntityByJavaId(byteBuffer.getLong()).getGeyserId());
        
        session.sendUpstreamPacket(emotePacket);
```